libsolidity/codegen/ContractCompiler.cpp

```diff
@@ -570,8 +570,9 @@ void ContractCompiler::appendReturnValuePacker(TypePointers const& _typeParamete
 
 void ContractCompiler::registerStateVariables(ContractDefinition const& _contract)
 {
-	for (auto const& var: ContractType(_contract).stateVariables())
-		m_context.addStateVariable(*std::get<0>(var), std::get<1>(var), std::get<2>(var));
+	for (auto const location: {DataLocation::Storage, DataLocation::Transient})
```

Comment:
[Ha, told you so :D](https://github.com/ethereum/solidity/pull/15247#discussion_r1689489140)

In hindsight, I should have probably pushed harder to have `.stateVariables(...)` return *all* state variables by default, and only allow filtering on a per need basis. After all, both storage *and* transient storage variables are state variables.

Reply:
I am now slightly concerned that we may have missed more cases like this one, even though the test coverage is now good :/